### PR TITLE
Extract build reaping out of enqueue_jobs.

### PR DESCRIPTION
**Pull request by fotinakis**
This should be a medium-large-ish performance improvement to the enqueue_jobs loop.

We happen to be doing reaping of the "active builds" list on every enqueue_jobs loop iteration. This can be extracted to sit as it's own side process instead of in the critical path.

Also: fix a bug where we were indexing over the active build list one at a time instead of loading the whole build list into memory. This could cause "build skipping" bugs when the active build list shifts underneath while the loop is running.

### CODE REVIEW

File: lib/percy/hub.rb [hunk 1]
```diff
@@ -363,17 +365,10 @@ def _enqueue_jobs
...
-          # ZRANGE returns items ordered by score. This ensures priority for older builds.
-          build_id = redis.zrange('builds:active', index, index).first
-          build_inserted_at = redis.zscore('builds:active', build_id)
-
-          # We've iterated through all the active builds and successfully checked and/or enqueued
-          # all potential jobs for all idle workers. Sleep for a small amount of time before
-          # checking again to see if locks have been released or idle capacity is available.
-          return 0.05 unless build_id
+        # ZRANGE returns items ordered by score. This ensures priority for older builds.
+        build_ids = redis.zrange('builds:active', 0, -1)
```
Comment (fotinakis) [inline lib/percy/hub.rb:369]: This fixes the potential "build skipping" bugs.

Comment (faun) [reply lib/percy/hub.rb:369]: These keys should all fit in memory just fine, right?

Comment (faun) [reply lib/percy/hub.rb:369]: These are just build IDs, of course they'll fit in memory.

**Review by faun — APPROVED** [review]
LGTM. 🌵

**Pull request merged**